internal/conduit/registry/geth_block.go

```diff
@@ -18,82 +21,201 @@ const (
 
 // GethBlockODef ...GethBlock register oracle definition used to drive oracle component
 type GethBlockODef struct {
-	cfg *config.OracleConfig
-	// TODO - Bind this to an interface and mock so that this logic can be tested
-	client *ethclient.Client
-
+	cfg        *config.OracleConfig
+	client     client.EthClientInterface
 	currHeight *big.Int
 }
 
 // NewGethBlockOracle ... Initializer
 func NewGethBlockOracle(ctx context.Context,
-	ot pipeline.OracleType, cfg *config.OracleConfig) (pipeline.Component, error) {
-	od := &GethBlockODef{cfg: cfg, currHeight: nil}
-
+	ot pipeline.OracleType, cfg *config.OracleConfig, client client.EthClientInterface) (pipeline.Component, error) {
+	od := &GethBlockODef{cfg: cfg, currHeight: nil, client: client}
 	return pipeline.NewOracle(ctx, ot, od)
 }
 
-func (oracle *GethBlockODef) ConfigureRoutine(ctx context.Context) error {
-	// TODO - Introduce starting block parameter
-	log := ctxzap.Extract(ctx)
+func (oracle *GethBlockODef) ConfigureRoutine() error {
+	ctxTimeout, ctxCancel := context.WithTimeout(context.Background(),
+		time.Second*time.Duration(models.EthClientTimeout))
+	defer ctxCancel()
+
+	log := ctxzap.Extract(ctxTimeout)
 	log.Info("Setting up GETH Block client")
-	client, err := ethclient.Dial(oracle.cfg.RPCEndpoint)
+
+	err := oracle.client.DialContext(ctxTimeout, oracle.cfg.RPCEndpoint)
+
 	if err != nil {
 		return err
 	}
-
-	oracle.client = client
 	return nil
 }
 
+// getCurrentHeightFromNetwork ... Gets the current height of the network and will not quit until found
+func (oracle *GethBlockODef) getCurrentHeightFromNetwork(ctx context.Context) *types.Header {
+	log := ctxzap.Extract(ctx)
+
+	for {
+		header, err := oracle.client.HeaderByNumber(ctx, nil)
+		if err != nil {
+			log.Error("problem fetching current height from network", zap.Error(err))
+			continue
+		}
+		return header
+	}
+}
+
 // BackTestRoutine ...
-func (oracle *GethBlockODef) BackTestRoutine(_ context.Context, _ chan models.TransitData) error {
-	// TODO - implement
+func (oracle *GethBlockODef) BackTestRoutine(ctx context.Context, componentChan chan models.TransitData,
+	startHeight *big.Int, endHeight *big.Int) error {
+	log := ctxzap.Extract(ctx)
+	if endHeight.Cmp(startHeight) < 0 {
+		return errors.New("start height cannot be more than the end height")
+	}
 
-	return nil
+	currentHeader := oracle.getCurrentHeightFromNetwork(ctx)
+
+	if startHeight.Cmp(currentHeader.Number) == 1 {
+		return errors.New("start height cannot be more than the latest height from network")
+	}
+
+	ticker := time.NewTicker(pollInterval * time.Millisecond)
+	height := startHeight
+
+	for {
+		select {
+		case <-ticker.C:
+
+			headerAsInterface, err := oracle.fetchData(ctx, height, models.FetchHeader)
+			headerAsserted, headerAssertedOk := headerAsInterface.(*types.Header)
+			// means the above retries failed
+			if err != nil || !headerAssertedOk {
+				log.Error("problem fetching header", zap.Error(err))
+
```

Comment:
In the case where `err = nil && headerAssertedOk == false`; we'd output:
```
{"msg": "problem fetching header", "error": nil }
```

Maybe want to add a specific message or error for a failed type assertion.

Reply:
Ah fixed it. Logged both probable errors since type assertion only returns bool.